lib/util/timestamp.ts

```diff
@@ -0,0 +1,75 @@
+import { DateTime } from 'luxon';
+
+export type Timestamp = string & { __timestamp: never };
+
+const timezoneOffset = new Date().getTimezoneOffset() * 60000;
+
+const millenium = 946684800000; // 2000-01-01T00:00:00.000Z
+const tomorrowOffset = 86400000; // 24 * 60 * 60 * 1000;
+
+function isValid(date: DateTime): boolean {
+  const tomorrow = DateTime.now().toMillis() + tomorrowOffset; // 24 * 60 * 60 * 1000;
+  const ts = date.toMillis();
+  return date.isValid && ts > millenium && ts < tomorrow;
+}
+
+export function asTimestamp(input: unknown): Timestamp | null {
+  if (input instanceof Date) {
+    const date = DateTime.fromJSDate(input, { zone: 'UTC' });
```

Comment:
don't we change the timezone to a probably wrong value here? or is datetime correctly converting it.

Reply:
It's converted from whatever timezone the original value is, to the UTC

Reply:
BTW this branch is unlikely to be used at all